python3/shipper/tests/shipper_tests.py

```diff
@@ -0,0 +1,152 @@
+import gzip
+import httpretty
+import json
+import os
+import unittest
+import urllib.request
+
+from python3.shipper.shipper import MaxRetriesException, UnauthorizedAccessException, UnknownURL
+from python3.shipper.shipper import LogzioShipper
+from io import BytesIO
+
+
+class TestLambdaFunction(unittest.TestCase):
+    def setUp(self):
+        # Set os.environ for tests
+        os.environ['URL'] = "https://listener.logz.io:8071"
+        os.environ['TOKEN'] = "12345678"
+        os.environ['TYPE'] = "test_log"
+        self._logzioUrl = "{0}/?token={1}".format(os.environ['URL'], os.environ['TOKEN'])
+        self._dec_data = []
+
+    @staticmethod
+    def generate_logs():
+        logs = [
+            {'k1': 'v1'},
+            {'k2': 'v2'},
+            {'k3': 'v3'},
+            {'k4': 'v4'}
+        ]
+        return logs
+
+    def ship_logs(self, logs):
+        shipper = LogzioShipper(self._logzioUrl)
+        for log in logs:
+            shipper.add(log)
+        shipper.flush()
+        return logs
+
+    @staticmethod
+    def delete_new_line(str_log):
+        if str_log.endswith('\n'):
+            return str_log[:-1]
+        return str_log
+
+    def validate_data(self, request, logs):
```

Comment:
The function's flow is hard to understand. Please extract to smaller functions

Reply:
Flow changed.